src/lib/containers/GenericCard.tsx

```diff
@@ -525,6 +535,30 @@ export default class GenericCard extends React.Component<
       }
     }
 
+    const fileHandleId = imageFileHandleIdValue || linkValue
+
+    /**
+     * To show a direct download link to a file, we need to determine the association that gives permission to download the file.
+     */
+    let fileHandleAssociation: FileHandleAssociation | undefined = undefined
+    if (table) {
+      if (isTableEntity(table)) {
+        // The file handle is in the table
+        fileHandleAssociation = {
+          fileHandleId,
+          associateObjectId: table?.id ?? '',
+          associateObjectType: FileHandleAssociateType.TableEntity,
+        }
+      } else {
+        // We're looking at a view, so the FileEntity (whose ID matches the row ID) gives permission to download the file handle
+        fileHandleAssociation = {
+          fileHandleId,
+          associateObjectId: data[schema.id],
+          associateObjectType: FileHandleAssociateType.FileEntity,
+        }
+      }
+    }
```

Comment:
Before, we would pass all of the information needed to construct this FileHandleAssociation object as props in FileHandleLink and ImageFileHandle components, including whether or not this is a table or a view.

Now, FileHandleLink and ImageFileHandle just take the association object. The logic related to table/view was moved here, so those components have APIs that make sense outside of a table query.